model/jpa/src/main/java/org/keycloak/organization/jpa/JpaOrganizationProvider.java

```diff
@@ -398,7 +398,7 @@ public boolean removeMember(OrganizationModel organization, UserModel member) {
 
     @Override
     public boolean isEnabled() {
-        return getAllStream().findAny().isPresent();
+        return realm.isOrganizationsEnabled();
```

Comment:
> This will change the current behavior when running the authenticator because the identity-first login page will be always shown regardless if there are organizations or not.

I see, thanks.

> Can we do this instead?

```
return realm.isOrganizationsEnabled() && getAllStream().findAny().isPresent();
```

While this would certainly solve aforementioned issue, it'd break the semantics of the `OrganizationProvide.isEnabled` method.

I believe we should solve this in a way:
- create util method within `Organizations` class, something like : `isOrganizationsPresent()` 
- update places where we need to check that apart the provider is enabled, also there are some organizations present with this new method

Also we may probably want to replace `getAllStream().findAny().isPresent()` with a `count` query, probably as a follow-up.

Reply:
Updated accordingly, created the follow-up: [#29697](https://github.com/keycloak/keycloak/issues/29697)